### PR TITLE
Add example `oauth2` and `InList`/`NotInList`

### LINKED ISSUE #250: Implement InList and NotInList methods
Some matcher structs (Number, DateTime) have `InRange` method that checks that the value belongs to given range `[min; max]`.

Another convenient thing would be `InList` method, that checks that the value equals to one of the given values `a, b, c, ...`.

For example:

```go
array.Element(1).String().InList("green", "red")
```

We need to add `InList` and counterpart `NotInList` methods to the following structs:

* Value
* Object
* Array
* String
* Number
* Boolean (just for completeness)
* Duration
* DateTime

The new methods can use assertions of type AssertBelongs and AssertNotBelongs. Please place these methods after `InRange`, or after `Equal*` if there is no `InRange` for this struct.

Each method should have documentation comment with code example and be covered by a unit test. It would be also nice to demonstrate usage of InRange and InList in `_examples/fruits_test.go`.

### PR DESCRIPTION
**Pull request by fahimbagar**
This PR does the following:
- Add new example for `InList` and `NotInList` inside `fruits.go`
- Add new example for `oauth2`
- Update README.md

Closes #250 and #251

### CODE REVIEW

**Comment by gavv** [comment]
Sorry, seems my recent commit created conflict. Could you please rebase on fresh master?

**Comment by fahimbagar** [comment]
It's done, @gavv.

File: _examples/oauth2.go [hunk 1]
```diff
@@ -0,0 +1,364 @@
...
+		func(ctx context.Context,
+			clientID,
+			username,
+			password string) (userID string, err error) {
+			if username == usernameConfig && password == passwordConfig {
+				userID = userIDConfig
+			}
+			return
+		},
+	)
```
Comment (gavv) [inline _examples/oauth2.go:63]: Since other handlers are extracted into functions, maybe extract this one too?

File: _examples/oauth2.go [hunk 2]
```diff
@@ -0,0 +1,364 @@
...
+	if _, ok := st.Get("LoggedInUserID"); !ok {
+		w.Header().Set("Location", "/login")
+		w.WriteHeader(http.StatusFound)
+		return
+	}
+
+	outputHTML(w, r, "static/auth.html")
+}
+
+func outputHTML(w http.ResponseWriter, req *http.Request, filename string) {
```
Comment (gavv) [inline _examples/oauth2.go:350]: Why do we need html pages for this example? Could we remove them?

**Review by gavv — COMMENTED** [review]
Thanks for PR! Here are a few comments.

File: _examples/oauth2.go [hunk 3]
```diff
@@ -0,0 +1,364 @@
...
+		e := json.NewEncoder(w)
+		e.SetIndent("", "  ")
+		_ = e.Encode(data)
+	})
+
+	return mux
+}
+
+// AuthClientHandler is a simple http.Handler that implements go-oauth2 client.
+func AuthClientHandler() http.Handler {
```
[the remark this answers is not in the document]
Comment (fahimbagar) [reply _examples/oauth2.go:141]: Ah, I misunderstood the intention. Let me rewrite it then.

File: _examples/oauth2.go [hunk 4]
```diff
@@ -0,0 +1,364 @@
...
+	srv.SetUserAuthorizationHandler(userAuthorizeHandler)
+
+	srv.SetInternalErrorHandler(func(err error) (re *errors.Response) {
+		log.Println("Internal Error:", err.Error())
+		return
+	})
+
+	srv.SetResponseErrorHandler(func(re *errors.Response) {
+		log.Println("Response Error:", re.Error.Error())
+	})
```
[the remark this answers is not in the document]
Comment (fahimbagar) [reply _examples/oauth2.go:74]: Noted, I'll remove unnecessary things here.

File: _examples/oauth2.go (hunk 2 above)
Comment (fahimbagar) [reply _examples/oauth2.go:350]: Noted, I'll remove unnecessary things here.

File: README.md [hunk 5]
```diff
@@ -120,6 +120,10 @@ See [`_examples`](_examples) directory for complete standalone examples.
 
     Testing with custom formatter for assertion messages.
 
+* [`oauth_test.go`](_examples/oauth2_test.go)
+
+  Testing a OAuth2 server with [`oauth2`](https://github.com/go-oauth2/oauth2/). Tests use either HTTP client or invoke the `http.Handler` directly.
```
[the remark this answers is not in the document]
Comment (fahimbagar) [reply README.md]: addressed in https://github.com/gavv/httpexpect/pull/267/commits/14c735f607039816f086b4530374cdacae7a1b18

File: _examples/fruits_test.go [hunk 6]
```diff
@@ -88,6 +88,8 @@ func TestFruits(t *testing.T) {
 	colors.Element(0).String().HasSuffix("een")
 
 	fruit.Value("weight").Number().IsEqual(200)
+	fruit.Value("weight").Number().InList(100, 200, 300)
+	fruit.Value("weight").Number().NotInList(400, 500, 600)
```
[the remark this answers is not in the document]
Comment (fahimbagar) [reply _examples/fruits_test.go]: addressed in https://github.com/gavv/httpexpect/pull/267/commits/ec52710a394530e6667463791f7bb1b6e78b8acc

File: _examples/oauth2.go (hunk 3 above)
[the remark this answers is not in the document]
Comment (fahimbagar) [reply _examples/oauth2.go]: Rewriting as simple as possible in https://github.com/gavv/httpexpect/pull/267/commits/58a65a21421eb8c9508f918706064fb15d6d5203.

Also throwing out unnecessary things.

File: _examples/oauth2.go (hunk 4 above)
[the remark this answers is not in the document]
Comment (fahimbagar) [reply _examples/oauth2.go]: Rewriting as simple as possible in https://github.com/gavv/httpexpect/pull/267/commits/58a65a21421eb8c9508f918706064fb15d6d5203.

Also throwing out unnecessary things.

File: _examples/oauth2.go (hunk 1 above)
Comment (fahimbagar) [reply _examples/oauth2.go]: Addressed

File: _examples/oauth2.go (hunk 2 above)
Comment (fahimbagar) [reply _examples/oauth2.go]: Addressed

**Comment by gavv** [comment]
Thank you!

**Comment by gavv** [comment]
Follow-up commit: f29b995bc36151192cf638804e6cf655a77981a4

* don't access client store from test; the only things shared between server and test are strings for client id and secret
* use hard-coded strings for simplicity
* cosmetic fixes

**Pull request merged**